create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/ScopusHandler.java

```diff
@@ -42,52 +45,69 @@ public ScopusHandler(S3Client s3Client) {
     @Override
     public CreatePublicationRequest handleRequest(S3Event event, Context context) {
         return attempt(() -> readFile(event))
-                .map(this::parseXmlFile)
-                .map(this::generateCreatePublicationRequest)
-                .orElseThrow(fail -> logErrorAndThrowException(fail.getException()));
+            .map(this::parseXmlFile)
+            .map(this::generateCreatePublicationRequest)
+            .orElseThrow(fail -> logErrorAndThrowException(fail.getException()));
     }
 
     private CreatePublicationRequest generateCreatePublicationRequest(DocTp docTp) {
         CreatePublicationRequest createPublicationRequest = new CreatePublicationRequest();
         createPublicationRequest.setAdditionalIdentifiers(generateAdditionalIdentifiers(docTp));
+        createPublicationRequest.setEntityDescription(generateEntityDescription(docTp));
         return createPublicationRequest;
     }
 
     private Set<AdditionalIdentifier> generateAdditionalIdentifiers(DocTp docTp) {
         return extractItemIdentifiers(docTp)
-                .stream()
-                .filter(this::isScopusIdentifier)
-                .map(this::toAdditionalIdentifier)
-                .collect(Collectors.toSet());
+            .stream()
+            .filter(this::isScopusIdentifier)
+            .map(this::toAdditionalIdentifier)
+            .collect(Collectors.toSet());
     }
 
     private List<ItemidTp> extractItemIdentifiers(DocTp docTp) {
         return docTp.getItem()
-                .getItem()
-                .getBibrecord()
-                .getItemInfo()
-                .getItemidlist()
-                .getItemid();
+            .getItem()
+            .getBibrecord()
+            .getItemInfo()
+            .getItemidlist()
+            .getItemid();
+    }
+
+    private EntityDescription generateEntityDescription(DocTp docTp) {
+        EntityDescription entityDescription = new EntityDescription();
+        entityDescription.setReference(generateReference(docTp));
+        return entityDescription;
+    }
```

Comment:
Det finnes en Builder-pattern i EntityDescription. 
something along the lines `entityDescription.withReference(ref).build()`
Tror det er smak og behag, men siden den er der ...